### PR TITLE
Added missing unittests for primitive schema types

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation
Currently we dont have any unittests for schemas. This pr adds them

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java [hunk 1]
```diff
@@ -146,6 +146,10 @@ public void close() throws Exception {
 
     private boolean isCompatible(SchemaAndMetadata existingSchema, SchemaData newSchema,
                                  SchemaCompatibilityStrategy strategy) {
+        if (SchemaType.isPrimitiveSchemaType(existingSchema.schema.getType())) {
+            // for primitive data types, only schema type check is needed
+            return existingSchema.schema.getType() == newSchema.getType();
```
Comment (jerrypeng) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java]: @srkukarni even from primitive types you still should check this condition:
```
compatibilityChecks.getOrDefault(newSchema.getType(), SchemaCompatibilityCheck.DEFAULT)
            .isCompatible(existingSchema.schema, newSchema, strategy);	            .isCompatible(existingSchema.schema, newSchema, strategy)
```

Comment (srkukarni) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java]: @jerrypeng fixed

**Comment by srkukarni** [comment]
@sijie @jerrypeng Based on @sijie feedback and our discussion, I've removed the schema changes and only kept the unittest additions. Please take a look again. Thanks!

**Comment by srkukarni** [comment]
run cpp tests
run java8 tests

**Pull request merged**